include/net/net_ip.h

```diff
@@ -870,6 +870,29 @@ char *net_addr_ntop(sa_family_t family, const void *src,
 		    char *dst, size_t size);
 
 /**
+ * @brief Parse a string that contains either IPv4 or IPv6 address
+ * and optional port, and store the information in user supplied
+ * sockaddr struct.
+ *
+ * @details Syntax of the IP address string:
```

Comment:
I'd say this should specify explicitly what will be on sockaddr if port is not specified (e.g. port field set to 0, or yet better not touched, to let the caller to initialize it to default value prior to call).